provisioning/atat_provisioning.yaml

```diff
@@ -134,52 +166,82 @@ paths:
           required: true
           schema:
             type: string
+        - name: environmentId
+          in: path
+          description: ID of the Environment
+          required: true
+          schema:
+            type: string
+      responses:
+        '200':
+          description: Successful operation
+          content:
+            application/json:
+              schema:
+                $ref: '#/components/schemas/EnvironmentOperatorRole'
+        '405':
+          description: Invalid input
       requestBody:
+        description: An EnvironmentOperatorRole that associates the Operator with a specific level of access
         required: true
         content:
           application/json:
             schema:
-              $ref: '#/components/schemas/CostRequest'
-            examples:
-              CostRequestActual:
-                $ref: '#/components/examples/CostRequestActual'
+              $ref: '#/components/schemas/EnvironmentOperatorRole'
+  '/portfolios/{portfolioId}/cost':
+    post:
+      tags:
+        - cost
+      summary: Returns cost data by Portfolio
+      description: Queries CSP for actual and forecasted cost data by Portfolio
+      operationId: getCostsByPortfolio
+      parameters:
+        - name: portfolioId
+          in: path
+          description: ID of the portfolio
+          required: true
+          schema:
+            type: string
+      requestBody:
+        required: true
+        content:
+          application/json:
+            schema:
+              $ref: '#/components/schemas/FOOBAR'
       responses:
         '200':
           description: Cost operation successful
           content:
             application/json:
               schema:
-                $ref: '#/components/schemas/CostResponse'
-              examples:
-                CostResponseActual:
-                  $ref: '#/components/examples/CostResponseActual'
+                $ref: '#/components/schemas/FOOBAR'
         '400':
           description: Invalid ID supplied
         '404':
-          description: Portfolio or Application not found
-  '/portfolios/{portfolioId}/applications/{applicationId}/environments/{environmentId}/cost':
+          description: Portfolio not found
+  '/portfolios/{portfolioId}/taskorders/{taskOrderNumber}/clins/{clin}/cost':
```

Comment:
There are two separate concepts here:

1.  Do we need to associate Task Orders / CLINs with Portfolios?
2. Who is responsible for determining which Task Orders are considered active at any given time?

I think the answer to the first question is Yes regardless of how we design this, which means that we're missing a few critical pieces here - the ability to associate task orders and CLINs to a Portfolio on initial provisioning as well as to add new ones. That would enable the operation you added.

The other question boils down to whether we can also have a GET `/portfolios/:id/cost` operation. If the CSPs are responsible, then we can; otherwise we can't. I lean towards the latter because it means we control that algorithm and provides a bit more stability. The only downside is that we have to invoke a few more operations in order to get the full cost data for a portfolio.

Reply:
Assuming CSPs are not responsible for determining which Task Orders are considered active at any given time and that we will internally call the cost operation for CLINs repeatedly.

Removed the GET `/portfolios/:id/cost` operation in 8f489e4771d11cf5b5f38f1773ac14bf4febd678

Reply:
To enabled operation _Add operation for Get actual & forecast spend by CLIN_, added two tasks to TODO list for associating Task Orders/CLINs.